src/DRICH_geo.cpp

```diff
@@ -546,6 +546,12 @@ static Ref_t createDetector(Detector& desc, xml::Handle_t handle, SensitiveDetec
 
     // END SENSOR MODULE LOOP ------------------------
 
+    // add constant for access to the number of modules per sector
+    if (isec == 0)
+      desc.add(Constant("DRICH_num_sensors", std::to_string(imod)));
+    else if (imod != desc.constantAsLong("DRICH_num_sensors"))
```

Comment:
Is there no deterministic variable that can be used? Does this need to be the final value of a loop counter?

Reply:
Not really, this comes from the spherical tiling algorithm above, with constraints set by https://github.com/eic/epic/blob/6446592846efd24d7d4c902b0f7b1e5e30970314/compact/pid/drich.xml#L197-L222
We have a few other derived 'constants', such as the mirror center and radius; the compact file specifies the desired focal region (offsets) and the plugin uses spherical optics equations to determine the mirror parameters from this specification.